torch/linalg/__init__.py

```diff
@@ -140,3 +140,51 @@
     >>> LA.norm(m[0, :, :]), LA.norm(m[1, :, :])
     (tensor(3.7417), tensor(11.2250))
 """)
+
+tensorsolve = _add_docstr(_linalg.linalg_tensorsolve, r"""
+linalg.tensorsolve(input, other, dims=None, *, out=None) -> Tensor
+
+Computes a tensor ``x`` such that ``tensordot(input, x, dims=x.ndim) = other``.
+The resulting tensor ``x`` has the shape equal to ``Q``, where ``Q`` is the shape
```

Comment:
"has the shape equal to Q" -> "has the same shape as Q"

Reply:
Before describing the shape of a result, I think this should start a new paragraph and discuss the shape requirement for input and other. That should make defining the shape of x easier. 

As I understand it, if the shape of input is (a, b, c, d, ... z), then the shape of other must (a, b, c, d...) and the shape of x will be (..., x, y z). Is that correct?

Reply:
> As I understand it, if the shape of input is (a, b, c, d, ... z), then the shape of other must (a, b, c, d...) and the shape of x will be (..., x, y z). Is that correct?

That is correct, but ...
Here is how I understand the current NumPy doc. First, the `other` tensor is fixed, that's why it can be of any shape. Then based on `other.shape` the requirements on `input` are posed. If `other.shape` is `(a, b, c, d)` then:
- `input.shape` must start with `a, b, c, d`
  - at this stage `torch.movedim(input, dims, (-(len(dims)+1), ..., -2, -1))` can be used to match the requirement.
 (moving dimensions specified in `dims` to the right end keeping the order)
- `input.shape = (a, b, c, d) + (..., x, y, z) = (a, b, c, d, ..., x, y, z)`
- `x.shape = (..., x, y, z)`

For the function to work `input` must satisfy the requirement `prod(..., x, y, z)==prod(a, b, c, d)`.

Alternatively, we can start with `input` and `other.ndim`, then:
- `other.shape` must be such that `other.shape == input.shape[0:other.ndim]` (that point is a bit problematic since we pose the requirement on `other` using `other.ndim`.
- `x.shape = input.shape[other.ndim:]`

`input` must satisfy the requirement `prod(input.shape[other.ndim:] == input.shape[:other.ndim]`.

I think that the second variant is better, because then we don't introduce any `Q`, and the "squareness" requirement is directly relatable to `tensorinv` function with `ind = other.ndim`.